msbuild/Xamarin.Shared/Xamarin.Shared.targets

```diff
@@ -0,0 +1,132 @@
+<!--
+***********************************************************************************************
+Xamarin.Shared.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+  created a backup copy.  Incorrect changes to this file will make it
+  impossible to load or build your projects from the command-line or the IDE.
+
+This file imports the version- and platform-specific targets for the project importing
+this file. This file also defines targets to produce an error if the specified targets
+file does not exist, but the project is built anyway (command-line or IDE build).
+
+Copyright (C) 2018 Microsoft. All rights reserved.
+***********************************************************************************************
+-->
+
+<!-- This is shared between Xamarin.iOS and Xamarin.Mac -->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Import Project="$(MSBuildThisFileDirectory)$(MSBuildThisFileName).Before.targets"
+			Condition="Exists('$(MSBuildThisFileDirectory)$(MSBuildThisFileName).Before.targets')"/>
+
+	<PropertyGroup>
+		<GetBindingProjectResourcesDependsOn>
+			_SplitProjectReferencesByFileExistence;
+		</GetBindingProjectResourcesDependsOn>
+	</PropertyGroup>
+
+	<!-- Invoked "recursively" by the App (before mmp\mtouch) which will fire on each library, so needs to be shared. -->
+	<Target
+		Name="GetBindingProjectResources"
+		Returns="@(AllBindingProjectResources)"
+		DependsOnTargets="$(GetBindingProjectResourcesDependsOn)">
+
+		<MSBuild
+			Projects="@(_MSBuildProjectReferenceExistent)"
+			Targets="GetBindingProjectResources"
+			BuildInParallel="$(BuildInParallel)"
+			Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform); %(_MSBuildProjectReferenceExistent.SetTargetFramework)"
+			Condition="'@(_MSBuildProjectReferenceExistent)' != '' and '%(_MSBuildProjectReferenceExistent.Private)' != 'false'"
+			ContinueOnError="$(ContinueOnError)"
+			SkipNonexistentTargets="true"
+			RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)">
+
+			<Output TaskParameter="TargetOutputs" ItemName="_AllBindingProjectResources"/>
+		</MSBuild>
+
+		<ItemGroup>
+			<_AllBindingProjectResources Condition="'$(NoBindingEmbedding)' == 'true'" Include="@(NativeReference->'%(FullPath)')">
+				<Assembly>$(TargetPath)</Assembly>
+			</_AllBindingProjectResources>
+		</ItemGroup>
```

Comment:
Generally you don't name ItemGroup items in the plural since they are assumed to be an item collection already.

Typical naming convention for this would just be `_BindingProjectResource`